### PR TITLE
Include all exporters from primary contrib into our collector-contrib

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Adds the remaining exporters from `https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter` to the collector-contrib module.

### CODE REVIEW

**Review by sblumenthal — APPROVED** [review]
LGTM for container-integrations files

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

**Review by vickenty — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Review by wiyu — APPROVED** [review]
👍 for processes

**Review by guyarb — APPROVED** [review]

File: cmd/trace-agent/test/testsuite/otlp_test.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ apm_config:
 		}
 		defer r.KillAgent()
 
-		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline cmd/trace-agent/test/testsuite/otlp_test.go:107]: Nit:
```suggestion
		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: cmd/trace-agent/test/testsuite/otlp_test.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ apm_config:
...
```
Comment (knusbaum) [inline cmd/trace-agent/test/testsuite/otlp_test.go:161]: Nit:
```suggestion
		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: cmd/trace-agent/test/testsuite/otlp_test.go [hunk 3]
```diff
@@ -253,7 +253,7 @@ apm_config:
...
```
Comment (knusbaum) [inline cmd/trace-agent/test/testsuite/otlp_test.go:256]:
```suggestion
		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: cmd/trace-agent/test/testsuite/otlp_test.go [hunk 4]
```diff
@@ -319,7 +319,7 @@ apm_config:
...
```
Comment (knusbaum) [inline cmd/trace-agent/test/testsuite/otlp_test.go:322]:
```suggestion
		conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: comp/otelcol/collector-contrib/impl/collectorcontrib.go [hunk 5]
```diff
@@ -343,13 +381,51 @@ func (c *collectorcontribImpl) OTelComponentFactories() (otelcol.Factories, erro
 	}
 
 	exporterList := []exporter.Factory{
-		// datadogexporter.NewFactory(),
```
Comment (knusbaum) [inline comp/otelcol/collector-contrib/impl/collectorcontrib.go:346]: Should we remove this? It's already mentioned as temporarily disabled above.

File: pkg/network/usm/testutil/grpc/client.go [hunk 6]
```diff
@@ -150,7 +150,7 @@ type Options struct {
 
 // NewClient returns a new gRPC client
 func NewClient(addr string, options Options, withTLS bool) (Client, error) {
-	gRPCOptions := []grpc.DialOption{grpc.WithBlock()}
+	gRPCOptions := []grpc.DialOption{grpc.WithBlock()} //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/network/usm/testutil/grpc/client.go:153]: Is this supposed to be here? `DialOption` is not deprecated and there's no `grpc.Dial` or `grpc.DialContext` here.

File: pkg/process/events/listener_linux.go [hunk 7]
```diff
@@ -50,7 +50,7 @@ func NewListener(handler EventHandler) (*SysProbeListener, error) {
 		return nil, errors.New("event_monitoring_config.socket must be set")
 	}
 
-	conn, err := grpc.Dial(socketPath, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, url string) (net.Conn, error) {
+	conn, err := grpc.Dial(socketPath, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, url string) (net.Conn, error) { //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/process/events/listener_linux.go:53]:
```suggestion
	conn, err := grpc.Dial(socketPath, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, url string) (net.Conn, error) { //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: pkg/process/metadata/workloadmeta/collector/process_test.go [hunk 8]
```diff
@@ -46,7 +46,7 @@ type collectorTest struct {
 func acquireStream(t *testing.T, port int) pbgo.ProcessEntityStream_StreamEntitiesClient {
 	t.Helper()
 
-	cc, err := grpc.Dial(fmt.Sprintf("localhost:%v", port), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.Dial(fmt.Sprintf("localhost:%v", port), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/process/metadata/workloadmeta/collector/process_test.go:49]:
```suggestion
	cc, err := grpc.Dial(fmt.Sprintf("localhost:%v", port), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: pkg/process/metadata/workloadmeta/grpc_test.go [hunk 9]
```diff
@@ -101,7 +101,7 @@ func TestStreamServer(t *testing.T) {
 	// Drop first cache diff before gRPC connection is created
 	<-extractor.ProcessCacheDiff()
 
-	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/process/metadata/workloadmeta/grpc_test.go:104]:
```suggestion
	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: pkg/process/metadata/workloadmeta/grpc_test.go [hunk 10]
```diff
@@ -179,7 +179,7 @@ func TestStreamServerDropRedundantCacheDiff(t *testing.T) {
 		Pid2: proc2,
 	})
 
-	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/process/metadata/workloadmeta/grpc_test.go:182]:
```suggestion
	cc, err := grpc.Dial(srv.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: pkg/process/metadata/workloadmeta/grpc_test.go [hunk 11]
```diff
@@ -378,7 +377,7 @@ func setupGRPCTest(t *testing.T) (*WorkloadMetaExtractor, *GRPCServer, *grpc.Cli
 	require.NoError(t, err)
 	t.Cleanup(grpcServer.Stop)
 
-	cc, err := grpc.Dial(grpcServer.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.Dial(grpcServer.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/process/metadata/workloadmeta/grpc_test.go:380]:
```suggestion
	cc, err := grpc.Dial(grpcServer.addr.String(), grpc.WithTransportCredentials(insecure.NewCredentials())) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```

File: pkg/proto/pbgo/core/api.pb.gw.go [hunk 12]
```diff
@@ -531,7 +531,7 @@ func RegisterAgentSecureHandlerServer(ctx context.Context, mux *runtime.ServeMux
 // RegisterAgentHandlerFromEndpoint is same as RegisterAgentHandler but
 // automatically dials to "endpoint" and closes the connection when "ctx" gets done.
 func RegisterAgentHandlerFromEndpoint(ctx context.Context, mux *runtime.ServeMux, endpoint string, opts []grpc.DialOption) (err error) {
-	conn, err := grpc.Dial(endpoint, opts...)
+	conn, err := grpc.Dial(endpoint, opts...) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/proto/pbgo/core/api.pb.gw.go:534]: These are generated files, so not sure we should be hand-modifying them.

File: pkg/security/agent/client.go [hunk 13]
```diff
@@ -205,7 +205,7 @@ func NewRuntimeSecurityClient() (*RuntimeSecurityClient, error) {
 				BaseDelay: time.Second,
 				MaxDelay:  time.Second,
 			},
-		}))
+		})) //nolint:staticcheck // TODO (ASC) fix grpc.Dial is deprecated
```
Comment (knusbaum) [inline pkg/security/agent/client.go:208]: Do we need 2 of these?

File: pkg/util/grpc/agent_client.go [hunk 14]
```diff
@@ -41,13 +41,13 @@ func getGRPCClientConn(ctx context.Context, ipcAddress string, cmdPort string, o
...
 	log.Debugf("attempting to create grpc agent client connection to: %s", target)
-	return grpc.DialContext(ctx, target, opts...)
+	return grpc.DialContext(ctx, target, opts...) //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
 }
 
 // defaultAgentClientDialOpts default dial options to the main agent which blocks and retries based on the backoffConfig
 var defaultAgentClientDialOpts = []grpc.DialOption{
 	grpc.WithConnectParams(grpc.ConnectParams{Backoff: defaultBackoffConfig}),
-	grpc.WithBlock(),
+	grpc.WithBlock(), //nolint:staticcheck // TODO (ASC) fix grpc.DialContext is deprecated
```
Comment (knusbaum) [inline pkg/util/grpc/agent_client.go:50]: This is not a `DialContext` or `Dial` call.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Mostly nitpicks.

File: cmd/trace-agent/test/testsuite/otlp_test.go (hunk 1 above)
Comment (dustmop) [reply cmd/trace-agent/test/testsuite/otlp_test.go]: Thanks for catching these mistakes, I have pushed a commit to fix the incorrect method names for deprecations.

File: comp/otelcol/collector-contrib/impl/collectorcontrib.go (hunk 5 above)
Comment (dustmop) [reply comp/otelcol/collector-contrib/impl/collectorcontrib.go:346]: Yes it should be removed. Adding it as a dependency pulls in most of the agent code, which is bad. Instead, there is a narrowing datadogexporter coming down the pipeline that will eventually be used instead.

File: pkg/network/usm/testutil/grpc/client.go (hunk 6 above)
Comment (dustmop) [reply pkg/network/usm/testutil/grpc/client.go]: `grpc.WithBlock` is deprecated, the message should have been concerning that instead.

File: pkg/proto/pbgo/core/api.pb.gw.go (hunk 12 above)
Comment (dustmop) [reply pkg/proto/pbgo/core/api.pb.gw.go]: Fair enough, removed.

File: pkg/security/agent/client.go (hunk 13 above)
Comment (dustmop) [reply pkg/security/agent/client.go]: Nope, this was a mistake, thanks for catching.

File: pkg/util/grpc/agent_client.go (hunk 14 above)
Comment (dustmop) [reply pkg/util/grpc/agent_client.go:50]: It should be `grpc.WithBlock`, which is deprecated.

**Review by knusbaum — APPROVED** [review]
Thanks for addressing those. I think there are a couple still there that I mentioned, but this LGTM besides that.

**Review by jeremy-hanna — APPROVED** [review]
👍 LGTM from ASC

**Review by dineshg13 — CHANGES_REQUESTED** [review]
Please don't remove 
`	"go.opentelemetry.io/collector/exporter/debugexporter"
	"go.opentelemetry.io/collector/exporter/otlpexporter"
	"go.opentelemetry.io/collector/exporter/otlphttpexporter"

**Comment by dustmop** [comment]
> Please don't remove ` "go.opentelemetry.io/collector/exporter/debugexporter" "go.opentelemetry.io/collector/exporter/otlpexporter" "go.opentelemetry.io/collector/exporter/otlphttpexporter"

Fixed by commit 9ba9f65683

**Review by dineshg13 — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]
LGTM for CI Experience files

**Comment by KSerrania** [comment]
The licences check is failing though:
```
license-detector -f json /tmp/tmpr2ffe07q
+ core,go.opentelemetry.io/collector/exporter/debugexporter,Apache-2.0,Copyright The OpenTelemetry Authors
+ core,go.opentelemetry.io/collector/exporter/debugexporter/internal/metadata,Apache-2.0,Copyright The OpenTelemetry Authors
+ core,go.opentelemetry.io/collector/exporter/otlphttpexporter,Apache-2.0,Copyright The OpenTelemetry Authors
+ core,go.opentelemetry.io/collector/exporter/otlphttpexporter/internal/metadata,Apache-2.0,Copyright The OpenTelemetry Authors
Licenses are not up-to-date.

Please run 'inv generate-licenses' to update LICENSE-3rdparty.csv.
```

**Pull request merged**